EIPS/eip-5750.md

```diff
@@ -8,103 +8,91 @@ status: Review
+1. Add a new method
+  * What will the method name be?
+  * What will the parameters be?
+  * How many use-cases does a given method signature support?
+  * Does this support off-chain signatures?
+2. Use one or more existing parameters, or add one or more new ones
+  * Should existing parameters be repurposed, or should more be added?
+  * How many parameters should be used?
+  * What are their sizes and types?
 
-First of all, there are multiple ways to achieve extensibility for existing functions.
+Standardizing how methods can be extended helps to answer these questions.
 
-1. _Add a new method_, then these questions need to be answered:
-
-- what will be the method name?
-- what will be the full list of parameters?
-- How many use-case do a given method name support?
-- Do/Will they support off-chain signatures?
-
-2. _Use an existing parameter(s)_: then these questions need to be answered:
-
-- will it be one parameter?
-- w- will it be multiple parameters?
-- what are their sizes and types?
-
-Standardizing how method will be extended helps answer this questions.
-
-In particular, this EIP aims at achieving maximum backward (e.g. [EIP-721](./eip-721.md), [EIP-1155](./eip-1155.md)) and future compatibility e.g. supporting commit-reveal feature for arbitrary method ([EIP-5732](./eip-5732.md)), or adding digital signatures alongside with a method call. The goal and motivation is to allow other implementers and EIPs to further depend on the compatibility based on the specifications this EIP so all compliant method interface automatically becomes eligible for future new behaviors.
+Finally, this EIP aims to achievie maximum backward and future compatibility. Many EIPs already partially support this EIP, such as [EIP-721](./eip-721.md) and [EIP-1155]. This EIP supports many use cases, from commit-reveal schemes ([EIP-5732](./eip-5732.md)), to adding digital signatures alongside with a method call. Other implementers and EIPs should be able to depend on the compatibility granted by this EIP so that all compliant method interfaces are eligible for future new behaviors.
 
 ## Specification
 
-The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119 and RFC 8174.
 
-1. Any compliant contract's compliant method MUST have a `bytes` (dynamic size) as its last parameter of the method.
+A method MUST be compliant if and only if for all of its overloads, it already has a `bytes` (dynamic size) as the last parameter that is not used for any functionality, or has another overload with the exact same parameters and a `bytes` (dynamic size) as the last parameter that is not used for any functionality:
```

Comment:
I feel this part is a bit hard to read. I will update it in a follow up commit

Reply:
I'll work on the phrasing.